src/Integration/Notifications/SonarQubeNotifications.cs

```diff
@@ -0,0 +1,211 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Timers;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.State;
+using SonarLint.VisualStudio.Integration.WPF;
+
+using CancellationTokenSource = System.Threading.CancellationTokenSource;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    [Export(typeof(ISonarQubeNotifications))]
+    internal class SonarQubeNotifications : ViewModelBase, ISonarQubeNotifications
+    {
+        private string text = "You have no events.";
+        private bool hasUnreadEvents;
+        private bool isVisible;
+        private bool isBalloonTooltipVisible;
+
+        private DateTimeOffset lastRequestDate = DateTimeOffset.MinValue;
+        private readonly ITimer timer;
+        private readonly IStateManager stateManager;
+        private readonly ISonarQubeServiceWrapper sonarQubeService;
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+        private bool isEnabled = true;
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage] // Do not unit test MEF constructor
+        internal SonarQubeNotifications(IHost host)
+            : this(host.SonarQubeService, host.VisualStateManager,
+                  new TimerWrapper { Interval = 10000 /* should be 60sec */ })
+        {
+        }
+
+        internal SonarQubeNotifications(ISonarQubeServiceWrapper sonarQubeService,
+            IStateManager stateManager, ITimer timer)
```

Comment:
Seems like I am the only one being surprised about taking a timer as parameter instead of a timer factory. Basically you are not in charge of the lifetime of the timer and someone can stop it from outside....Weird....